Improve error handling

Right now errors seem not to be printed correctly, and cargo-sweep exits with code 0 even on error.

```text
$ cargo sweep --toolchains nightly-2019-13-13 # toolchain that does not exist
[INFO] Using specified installed toolchains: ["nightly-2019-13-13"]
[ERROR] Failed to clean "/project/target":
```

--- a/src/fingerprint.rs
+++ b/src/fingerprint.rs
@@ -303,7 +303,11 @@ fn lookup_from_names<'a>(
         let out = Command::new("rustc").args(args).output()?;
 
         if !out.status.success() {
-            bail!(String::from_utf8_lossy(&out.stdout).to_string());
+            if !out.stdout.is_empty() {
+                bail!(String::from_utf8_lossy(&out.stdout).to_string());
+            } else {
+                bail!(String::from_utf8_lossy(&out.stderr).to_string());
+            }
         }
         Ok(hash_u64(&String::from_utf8_lossy(&out.stdout)))
     })

